man/lxqt-sudo.1
.TH lxqt-sudo 1 "" "" "LXQt\ Module"
.SH NAME
\fBlxqt-sudo\fR, \fBlxsu\fR, \fBlxdoas\fR \- execute a command as privileged user
.SH SYNOPSIS
\fBlxqt-sudo\fR \fIoption\fR [\fIcommand\fR [\fIarguments\fR]]
.br
\fBlxsu\fR [\fIoption\fR] [\fIcommand\fR [\fIarguments\fR]]
.br
\fBlxsudo\fR [\fIoption\fR] [\fIcommand\fR [\fIarguments\fR]]
.br
\fBlxdoas\fR [\fIoption\fR] [\fIcommand\fR [\fIarguments\fR]]
.SH DESCRIPTION
\fBlxqt-sudo\fR (and symlinks \fBlxsu\fR, \fBlxsudo\fR, \fBlxdoas\fR) is a graphical QT frontend for plain \fBsudo(8)\fR, \fBsu(1)\fR or \fBdoas(1)\fR (for requesting optional password in GUI fashion).
.br
When invoked it simply spawns child \fIsudo\fR, \fIsu\fR or \fIdoas\fR process with requested \fIcommand\fR (and optional \fIarguments\fR). If \fIsudo\fR/\fIsu\fR/\fIdoas\fR requests user's password,
the GUI password dialog is shown and (after submit) the password is provided to backend.
.br
.SH OPTIONS
\fBoption\fR is one of:
.br
  -h|--help      Print help.
.br
  -v|--version   Print version information.
.br
  -q|--quiet     Make \fBlxqt-sudo\fR less verbose.
.br
  -s|--su        Use \fIsu\fR as backend (default for \fBlxqt-sudo\fR & \fBlxsudo\fR is \fIsudo\fR, for \fBlxsu\fR \fIsu\fR, for \fBlxdoas\fR \fIdoas\fR).
.br
  -d|--sudo      Use \fIsudo\fR as backend (default for \fBlxqt-sudo\fR & \fBlxsudo\fR is \fIsudo\fR, for \fBlxsu\fR \fIsu\fR, for \fBlxdoas\fR \fIdoas\fR).
.br
  -a|--doas      Use \fIdoas\fR as backend (default for \fBlxqt-sudo\fR & \fBlxsudo\fR is \fIsudo\fR, for \fBlxsu\fR \fIsu\fR, for \fBlxdoas\fR \fIdoas\fR).
.SH "REPORTING BUGS"
Report bugs to https://github.com/lxqt/lxqt/issues
.SH "SEE ALSO"
\fBsudo(8)\fR \fBsu(1)\fR \fBdoas(1)\fR
.SH AUTHOR
This manual page was created by \fBPalo Kisa\fR \fI<palo.kisa@gmail.com>\fR
for \fBLXQt\fR project.


CHANGELOG
lxqt-sudo-2.2.0 / 2025-04-17
=============================
 * Updated translations and bumped the version.

lxqt-sudo-2.1.0 / 2024-11-05
=============================
 * Add new env vars for `sudo`.

lxqt-sudo-2.0.0 / 2024-04-17
=============================
 * Allow bidirectional communication with launched process.
 * Avoid flushing empty line.
 * Flush everything comming from child and optimize splitting of output.
 * Ported to Qt6.

lxqt-sudo-1.4.0 / 2023-11-05
=============================
 * Updated translations and bumped the version.

lxqt-sudo-1.3.0 / 2023-04-15
=============================
 * Added support for doas(1) backend.

lxqt-sudo-1.2.0 / 2022-11-05
=============================
 * Updated translations and bumped the version.

lxqt-sudo-1.1.0 / 2022-04-15
=============================
 * Updated translations and bumped the version.

lxqt-sudo-1.0.0 / 2021-11-04
=============================
 * Updated AUTHORS.
 * Bumped minimum required Qt version to 5.15.

lxqt-sudo-0.17.0 / 2021-04-15
=============================
  * Handled Qt5.15's deprecations.

lxqt-sudo-0.16.0 / 2020-11-01
=============================
  * Use QFlags default constructor.

lxqt-sudo-0.15.0 / 2020-04-23
=============================
  * sudo: Fix (re)setting HOME with sudo backend.
  * sudo: Add TERM to allowed/preserved env vars.
  * Build on NetBSD.

lxqt-sudo-0.14.1 / 2019-02-24
=============================

  * Dont strip XAUTHLOCALHOSTNAME, it is required for lxqt-sudo to function
    properly.
  * Updated translations
  * Note for packagers: lxqt-sudo now depends on liblxqt >= 0.14.1

lxqt-sudo-0.14.0 / 2019-01-25
=============================

  * Implementation fixes:
    - sudo: Force "C" locale for su/sudo
    - sudo: Prevent malicious attempts for command injection
    - sudo: Use better child ending logic
    - sudo: Strip environment - Leave only required environment
      variables (for X & locale) to get into the elevated child
      process.
    - sudo: Make critical error messages less obscure
    - ui: Fixed the layout and line-break
  * Improved cmake scripting
    - Set cmake_minimum_required to 3.1.0
  * Moved translations from lxqt-l10n back to lxqt-sudo
    - Removed obsolete translation fuctionality
    - Added translation promo in README.md
  * Translation updates

lxqt-sudo-0.13.0 / 2018-05-21
=============================

  * passworddialog.cpp: include QIcon explicitely
  * CMake: Prevent in-source builds
  * fix http -> https
  * Fixed mentions of LXDE

lxqt-sudo-0.12.0 / 2017-10-21
=============================

  * Release 0.12.0: Update changelog
  * Set informal patch version
  * Added basic .gitattributes
  * Drops Qt5Core_VERSION_STRING
  * set Qt::AA_UseHighDpiPixmaps to true
  * Bump year

lxqt-sudo-0.11.1 / 2017-01-01
=============================

  * Release 0.11.1: Update changelog
  * Make CMakeLists.txt more uniform.

lxqt-sudo-0.11.0 / 2016-09-24
=============================

  * Release 0.11.0: Add changelog
  * Add README.md
  * build: Forward translations parameters
  * ts-files removal (#21)
  * Remove translations (will be pulled in build time)
  * Bump year
  * Italian translation update
  * Updated Russian translation Fix language and typos
  * Add Russian translation

lxqt-sudo-0.10.0 / 2015-11-02
=============================

  * Fix LICENSE file
  * Fix starting pcmanfm-qt
  * Preserve environment with sudo backend
  * Update Slovak translation
  * Updated translation template, updated german translation.
  * Add a plus to AUTHORS license AUTHORS must match the files fixes lxde/lxqt/issues/784 fix typo choosen --> chosen
  * lxqt-sudo: partial support for running cmd with arguments using su backend
  * lxqt-sudo: unified processing logic for su & sudo into common code
  * lxqt-sudo: lxsu & lxsudo symlinks for lxqt-sudo
  * lxqt-sudo: added support for su(1) backend
  * Add Polish translation
  * Rename LxQt to LXQt everywhere
  * Remove the country specific from language code.
  * Add Greek (el) translation
  * Created italian translation, fixed missing line <TS version="2.1" language="XX"> for _de.ts and -sk.ts
  * Traditional Chinese translation
  * Handles CMake policy CMP0063
  * Use the LXQtCompilerSettings CMake module
  * main: removed old useless define
  * Adds Runtime install COMPONENT
  * Coding style change
  * Use target_compile_definitions()
  * Use GNUInstallDirs
  * Use CMAKE_AUTOUIC, drop qt5_wrap_ui()
  * Use CMAKE_AUTOMOC, drop qt5_wrap_cpp()
  * Use CMAKE_INCLUDE_CURRENT_DIR
  * main: use standard liblxqt unix signal handling to quit
  * Updates the build system to the Targets infrastructure
  * German translation
  * lxqt-sudo: licensing

lxqt-sudo-0.9.0 / 2015-06-10
============================

  * lxqt-sudo: direct communication with sudo
  * lxqt-sudo: use dialog-password icon
  * lxqt-sudo: help & version info into stdout (instead of stderr)
  * man: bug reporting link fixed
  * lxqt-sudo: version parameter support
  * lxqt-sudo: added sk translation
  * lxqt-sudo: proper communication error messages to stderr
  * lxqt-sudo: removed .desktop file
  * lxqt-sudo: showing big "security-high" icon
  * lxqt-sudo: help/usage added
  * lxqt-sudo: information about sudo's failure
  * lxqt-sudo: manual page
  * lxqt-sudo: do not use -b for sudo (fixes some application starting problems)
  * lxqt-sudo: new GUI frontend for sudo


translations/lxqt-sudo.ts
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE TS>
<TS version="2.1">
<context>
    <name>PasswordDialog</name>
    <message>
        <location filename="../passworddialog.ui" line="6"/>
        <source>LXQt sudo</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../passworddialog.ui" line="42"/>
        <source>Copy command to clipboard</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../passworddialog.ui" line="45"/>
        <source>&amp;Copy</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../passworddialog.ui" line="83"/>
        <source>The requested action needs administrative privileges.&lt;br&gt;Please enter your password.</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../passworddialog.ui" line="106"/>
        <source>LXQt sudo backend</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../passworddialog.ui" line="109"/>
        <source>A program LXQt sudo calls in background to elevate privileges.</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../passworddialog.ui" line="119"/>
        <source>Command:</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../passworddialog.ui" line="126"/>
        <source>Password:</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../passworddialog.ui" line="133"/>
        <source>Enter password</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../passworddialog.cpp" line="60"/>
        <source>Attempt #%1</source>
        <translation type="unfinished"></translation>
    </message>
</context>
<context>
    <name>QObject</name>
    <message>
        <location filename="../sudo.cpp" line="78"/>
        <source>Usage: %1 option [command [arguments...]]

GUI frontend for %2/%3/%4

Arguments:
  option:
    -h|--help      Print this help.
    -v|--version   Print version information.
    -q|--quiet     Make %1 less verbose.
    -s|--su        Use %3(1) as backend.
    -d|--sudo      Use %2(8) as backend.
    -a|--doas      Use %4(1) as backend.
  command          Command to run.
  arguments        Optional arguments for command.

</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sudo.cpp" line="97"/>
        <source>%1 version %2
</source>
        <translation type="unfinished"></translation>
    </message>
</context>
<context>
    <name>Sudo</name>
    <message>
        <location filename="../sudo.cpp" line="211"/>
        <source>%1: no command to run provided!</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sudo.cpp" line="218"/>
        <source>%1: no backend chosen!</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sudo.cpp" line="235"/>
        <source>Syscall error, failed to fork: %1</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sudo.cpp" line="263"/>
        <source>unset</source>
        <extracomment>shouldn&apos;t be actually used but keep as short as possible in translations just in case.</extracomment>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sudo.cpp" line="329"/>
        <source>%1: Detected attempt to inject privileged command via LC_ALL env(%2). Exiting!
</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sudo.cpp" line="371"/>
        <source>Syscall error, failed to bring pty to non-block mode: %1</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sudo.cpp" line="379"/>
        <source>Syscall error, failed to fdopen pty: %1</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sudo.cpp" line="348"/>
        <source>%1: Failed to exec &apos;%2&apos;: %3
</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sudo.cpp" line="413"/>
        <source>Child &apos;%1&apos; process failed!
%2</source>
        <translation type="unfinished"></translation>
    </message>
</context>
</TS>
